test/controllers/users_controller_test.rb

```diff
@@ -1,39 +1,155 @@
 require 'test_helper'
 
 class UsersControllerTest < ActionController::TestCase
-  test "should get create" do
-    get :create
-    assert_response :success
+
+  ##########
+  #create tests
+  test "creating the same username or email again return unprocessable_entity" do
+    post :create, user: {
+                username: "bob",
+                name: "Bob",
+                email: "fake@fake.fake",
+                terms_and_conditions: true,
+                password: "Password1"}
+    assert_response :unprocessable_entity,
+                    "How is bob processable when bob is a fixture"
+  end
+
+  test "creating a user returns :found and saves user in the database" do
+    post :create, user: {
+                username: "sam",
+                name: "Sam",
+                email: "sam@sam.sam",
+                terms_and_conditions: true,
+                password: "Password1"}
+    assert_response :found, "\nFailed at creating Sam"
+    assert User.exists?(username: "sam"), "\nFailed at saving Sam"
   end
 
-  test "should get new" do
+  ##########
+  #new tests
+  test "new succeeds if user is not signed in and redirects to home if user is" do
     get :new
-    assert_response :success
+    assert_response :success or
+    (assert_redirected_to root_path,
+                    "User is signed in but failed at redirecting to home"
+     assert_response :found)
+  end
+
+  ##########
+  #likes tests
+  test "should be able to get likes through controller" do
+    @user = users(:bob)
+    get :likes, username: @user.username
+    assert_response :found
   end
 
-  test "should get edit" do
-    get :edit
-    assert_response :success
+  ##########
+  #additional_info tests
+  test "should be able to get additional_info" do
+    @user = users(:bob)
+    get :additional_info, username: @user.username, user: {}
+    assert_response :found
   end
 
-  test "should get update" do
-    get :update
-    assert_response :success
+  test "should be able to patch additional_info" do
+    post :create, user: {
```

Comment:
It's probably simpler to use an existing user like `:bob` rather than creating a new one.

Creating a new user adds extra steps to the test that might go wrong